### DISCUSSION TITLE
Theme is not fully functional on Rofi 1.7.0

### DISCUSSION THREAD

**Discussion by jpmvferreira** [davatorium/rofi#1441] [Theming] [answered]
Before the update my rofi sidebar looked like this:
![image](https://user-images.githubusercontent.com/41581666/132234431-58dea1d3-96ff-4be6-a0e3-2f0c950e5ebe.png)

Now I did a few minor changes with time, but aren't that relevant. Now after the update, and after some messing around in my config files, the theme currently looks like this:
![image](https://user-images.githubusercontent.com/41581666/132234541-6f6a7188-8ddf-4d36-a3e5-70ae48a1a334.png)

Rofi is now being executed with `rofi -show drun -modi drun -drun-display-format {name}`.

In `~/.config/rofi` I have two files: `config.rasi` and `sidebar.rasi`:
<details>
<summary>config.rasi</summary>
  
```
// Notes:
// - CSS hex colors format: #RRGGBBAA ==> normal hex + 2 digits for transparency

/*
Transparency digits:
100% — FF
95% — F2
90% — E6
85% — D9
80% — CC
75% — BF
70% — B3
65% — A6
60% — 99
55% — 8C
50% — 80
45% — 73
40% — 66
35% — 59
30% — 4D
25% — 40
20% — 33
15% — 26
10% — 1A
5% — 0D
0% — 00
*/

configuration {
    modi: "drun";
    show-icons: false;
}

// import my theme
@import "sidebar.rasi"

* {
    lines: 15;
    columns: 1;
    font: "SF Pro Text 12";

    // Background (CSS)
    background: #2A3440BA;

    // Input box (CSS)
    background-alt: #5a81b080;

    // ?? (CSS)
    border-color: #00000000;

    // Border color (hex)
    primary: #009FFF;

    // Text color (hex)
    text-color: #D8DEE9;

    background-color: transparent;
    background-color: @background;
}

#window {
    border:           0;
    border-radius:    4px;
    padding:          0;
    width:            20%;
    height:           90%;
    transparency:     "real";
    background-color: @background;
}

#mainbox {
    children: [entry, listview];
    spacing: 40px;
    /*margin: 20%;*/
    padding: 20px 40px;
}

#listview {
    background-color: transparent;
    fixed-height: 0;
    border: 0px;
    spacing: 5px;
    scrollbar: false;
    padding: 5px 0px 0px;
}

#element {
    border: 0;
    border-radius: 15px;
    padding: 10px;
    orientation: horizontal;
    background-color: transparent;
}

element normal.urgent, element alternate.urgent {
    background-color:               @urgent;
}

element normal.active, element alternate.active {
    background-color:               @background-alt;
    border:                         4px 4px solid 4px 4px;
    border-radius:                  8;
    border-color:                   transparent;
}

#element selected {
    border:                         2px 2px solid 2px 2px;
    border-radius:                  8;
    border-color:                   @primary;
}

element selected.active {
    background-color:               @background-alt;
}

element-icon {
    border:                         0px;
    border-color:                   @primary;
    border-radius:                  0;
    padding:                        0px;
}

#case-indicator {
    background-color: #00000000;
    spacing:    0;
}

#entry {
    expand:                         false;
    margin:                         0px 0px;
    background-color:               @background-alt;
    padding:                        12px;
    border:                         3px;
    border-color:                   @background-alt;
    border-radius:                  8px;
}

#prompt {
    background-color: #00000000;
    spacing:    0;
}

#textbox-prompt-colon {
    background-color: #00000000;
    expand:     true;
    str:        "";
    margin:     0px 0.0em 0em 0em ;
}

inputbar {
    background-color: transparent;
}
```
</details>

<details>
<summary>sidebar.rasi</summary>

```
configuration {
    theme:                          "config";
    window-format:                  "{t}";
    show-icons:                     true;
    font:                           "DejaVu Sans 13";
    columns:                        1;
}

window {
    height:                         912px;
    width:                          380px;
    location:                       west;
    anchor:                         west;
    x-offset:                       48;
    y-offset:                       0;
}

mainbox {
    border:                         2px;
    border-color:                   @primary;
    padding:                        24px;
}

element-icon {
    size: 32; /* Must be a power of 2 */
}

element-text {
    vertical-align: 0.5;
}
```
</details>

As you can see there's some dirty workarounds here, as I am not even sure where did I got this theme, and how did I modify it. All I did was add the `@import "sidebar.rasi"` to `config.rasi`, which breaks if placed somewhere else in the code expects right after the `configuration` entry. 
Even after that the window size was wrong and forced me to manually  change it in the `config.rasi` instead of being dictated by `sidebar.rasi` as it should be.
There's also the issue that the background colors for icons and text is not fully transparent, and I can't seem to fix that.

Other than that the other solutions provided in similar threads and in the release section showed no improvement and  would result in a white square with the correct placing, or something even weirder.

**Comment by jpmvferreira** [comment, marked as answer]
I should also note that this theme was written probably 2 years ago or more, I'm not even sure what version of rofi was this written on.

**Reply by DaveDavenport** [reply]
Please check the release notes or the [sticky](https://github.com/davatorium/rofi/issues/1397), I think the solution for the transparent elements is mentioned in there.

Try adding this to your theme:
```css
element-text, element-icon {
    background-color: inherit;
    text-color:       inherit;
}
```

**Reply by DaveDavenport** [reply]
The parsing goes as follow:
it loads the default theme.
When it encounters @import "sidebar.rasi"  it appends new/ overwrite existing settings to the default theme
then everything below the "sidebar.rasi"  is appended new/overwrite existing again.

So the `window { width: }`  below the import will overwrite the ` window { width: }`  set in the import "sidebar".

**Reply by DaveDavenport** [reply]
copy pasta:

`@theme` discard all theme options and load new one.
`@import` appends to the current theme.
Adding to the bottom of your config.rasi appens to the current theme.

By default the default rofi theme is loaded. (rofi -no-config -dump-theme to see it )

To clear this you can either put your theme in separate file and load it with @theme or

use `@theme "/dev/null"`.

**Reply by DaveDavenport** [reply]
hope this info helps fixing it.

**Reply by jpmvferreira** [reply]
Thanks, that did in fact fix the transparency issue.

As for the theme, I think what happens is that this is all over the place, so I'll probably just merge the two files into one and use that one file as my default theme, since this is the only one I'm using anyways.

There's still a few things for me to understand but it's working out as expected, so I'll save that for latter on.

Thanks again!

**Reply by jpmvferreira** [reply]
Also, this was my first time posting here I thought this was similar to the Github issue section, but it isn't. 

For future users, the solution was to add the first suggested block of code.

**Reply by DaveDavenport** [reply]
Can you mark the question answered (if it is).

**Reply by DaveDavenport** [reply]
( nice wallpaper, do you have a link?)

**Reply by jpmvferreira** [reply]
I tracked it down to here: https://www.artstation.com/badriel

I'm unsure where I got it from, but this seems to be the original source.

**Reply by DaveDavenport** [reply]
thanks.
